### PR TITLE
Added No Internet Screen At Root Level

### PR DESCRIPTION
**Pull request by ReddyGona**
https://github.com/saigopin/flutter_clean_architecture/assets/66050989/b536ddff-16ef-4ab1-97cf-57a8ecc5225b

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -44,33 +44,67 @@ void main() async {
...
-            ),
-          );
-        },
-      ),
+      child: const HomeScreen(),
+    );
+  }
+}
+
+class HomeScreen extends StatefulWidget {
```
Comment (narimetisaigopi) [inline lib/main.dart:63]: Why this code in home screen bro?

File: lib/src/core/styles/styles.dart [hunk 2]
```diff
@@ -17,6 +17,8 @@ mixin ButtonStyles {
     required bool showBorder,
     Color? disabledBackgroundColor,
     Color? disabledTextColor,
+    EdgeInsets? padding =
```
Comment (narimetisaigopi) [inline lib/src/core/styles/styles.dart:20]: Can you remove ? Here?

File: lib/src/core/styles/styles.dart [hunk 3]
```diff
@@ -38,6 +40,8 @@ mixin ButtonStyles {
     bool isButtonDisabled = false,
     Color? disabledBackgroundColor,
     Color? disabledTextColor,
+    EdgeInsets? padding =
```
Comment (narimetisaigopi) [inline lib/src/core/styles/styles.dart:43]: Same here

File: lib/src/shared/presentation/pages/no_internet_screen.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              buttonHeight: 80,
+              buttonWidth: 150,
+              child: InkWell(
+                onTap: () => context.read<NetworkCubit>().checkConnectivity(),
+                child: const Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: <Widget>[
+                    Icon(Icons.refresh, size: 30),
+                    Text(
+                      'Refresh',
```
Comment (narimetisaigopi) [inline lib/src/shared/presentation/pages/no_internet_screen.dart:49]: Get this from app strings

File: lib/src/shared/presentation/widgets/custom_button_widget.dart [hunk 5]
```diff
@@ -12,6 +12,7 @@ class CustomButtonWidget extends StatelessWidget with ButtonStyles {
   final Color? disabledTextColor;
   final bool isButtonDisabled;
   final bool showBorder;
+  final EdgeInsets? padding;
```
Comment (narimetisaigopi) [inline lib/src/shared/presentation/widgets/custom_button_widget.dart:15]: Why ? This require?

File: lib/main.dart (hunk 1 above)
Comment (ReddyGona) [reply lib/main.dart:63]: @saigopin we need dispose method, when ever the app reloads.

File: lib/src/shared/presentation/widgets/custom_button_widget.dart (hunk 5 above)
Comment (ReddyGona) [reply lib/src/shared/presentation/widgets/custom_button_widget.dart:15]: to controll the padding of button from out side

**Pull request merged**